### ISSUE TITLE
macOS Catalina 

### ISSUE THREAD

**Issue by bmw417** [img2tab/macos-guest-virtualbox#70]
Hey again,

Foxlet made a Catalina VM for QEMU here: https://github.com/foxlet/macOS-Simple-KVM

However, his implementation is a little clunky because of a hardcoded low resolution and hardcoded system version of 2009 iMac. I have a new 12.9" iPad Pro that should be able to run sidecar with Catalina, but because of the hardcoded system version it is incompatible with Catalina and sidecar. If only I could figure out what url to put in the $urlbase variable, I'd be able to do this (since he also has a script that generates a BaseSystem.img for Catalina). 

Any help on this would be awesome!

**Comment by img2tab** [comment]
This is certainly possible right now with only one change to the script. Before line 544, add:
```urlbase='http://swcdn.apple.com/content/downloads/25/17/041-71284/6n9ob65kcsjccw19npm020cuyf5zl1dfom/'```

This method is not robust and will break whenever macOS Catalina updates

The reason for using the swcdn URL and not the software update catalog file (`https://swscan.apple.com/content/catalogs/others/index-10.15-10.14-10.13-10.12-10.11-10.10-10.9-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog`) is that the macOS Catalina recovery environment does not have a working binary of `curl` or any other download software other than Safari, which requires manual interaction.

A different solution is [using a `VISO` file and downloading all the requirements to the host.](https://github.com/img2tab/macos-guest-virtualbox/issues/55#issuecomment-491796482) This requires VirtualBox ≥ 6, so I'll have to think about the implications of requiring VBox 6 instead of 5.2 for a bit, but that's likely where the script is heading since it'll solve a couple of other issues too.

**Comment by bmw417** [comment]
I'll try and run the script and report how it goes. I'm not too familiar with the inner workings of VBox, but I trust your judgement 100%. VBox 6.0 has to be at least 6 months old or so if memory serves me, so I don't think it would be a huge deal to require >= 6.0 moving forward. If someone was in a legacy environment and needed < 6.0 for whatever reason, I doubt there'd be a huge demand for the latest macOS anyway. On a different note, that's kinda crappy that curl is not shipped anymore with recovery, curl / wget are very basic tools I'd want there.

My method I'll try is to use foxlet's BaseSystem.img and then run your script on top of that to see if I can get Catalina up and running. Assuming I do, I'll turn off automatic updates to avoid breakage.

**Comment by bmw417** [comment]
I see what you mean by curl being messed with in the Recovery environment. The script fails at the 

`kbstring='urlbase="'"${urlbase}"'"; for filename in BaseSystem.chunklist InstallInfo.plist AppleDiagnostics.dmg AppleDiagnostics.chunklist BaseSystem.dmg InstallESDDmg.pkg; do curl -C - "${urlbase}${filename}" -o "/Volumes/'"${vmname}"'/${filename}"; done'
`

line. I ran curl by itself and what it returned was

```
dyld: lazy symbol binding failed: can't resolve symbol _OPENSSL_load_builtin_modules in /usr/lib/libcurl.4.dylib because dependent dylib #3 could not be loaded
dyld: dyld cache load error: shared cache file open() failed
dyld: can't resolve symbol _OPENSSL_load_builtin_modules in /usr/lib/libcurl.4.dylib because dependent dylib #3 could not be loaded
Abort trap: 6
```

I'm assuming there's a header file missing that would link where someone might've put a `#define _OPENSSL_load_builtin_modules var` but I don't have any macOS fu, just Linux.

**Comment by img2tab** [comment]
[The alternate script I posted works and installs properly.](https://github.com/img2tab/macos-guest-virtualbox/files/3295572/macos-guest-virtualbox.log)

**Comment by bmw417** [comment]
Worked perfect on Ubuntu 19.04, thanks!

**Comment by img2tab** [comment]
Keeping open since this feature will be implemented in the future.

**Comment by img2tab** [comment]
Implemented in [version 0.65.0](https://github.com/img2tab/macos-guest-virtualbox/releases/tag/v0.65.0)
